plugins/module_utils/network/ios/argspec/bgp_global/bgp_global.py

```diff
@@ -500,7 +500,7 @@ class Bgp_globalArgs(object):  # pylint: disable=R0903
                             "type": "dict",
                             "options": {
                                 "set": {"type": "bool"},
-                                "number": {"type": "int"},
+                                "number": {"type": "str"},
```

Comment:
The change looks valid. Could you add an action state as a test and include an integration test as well? We need to ensure that existing playbooks continue to function as expected. Please try passing the value as an integer in the playbook to verify that this change doesn't impact the current way of usage.

Reply:
Sure will do that thanks

Reply:
I have updated the unit test and added the required integration test in the new commit. I also tested the playbook by passing the value as an integer, and it worked correctly.

Reply:
can you add a test passing it an int value and one more passing it a float value, We need to ensure that it works,  allowing users to continue using it the same way as before. If existing playbooks work seamlessly after the change, we can avoid a deprecation process. However, if they don’t, we’ll need to find a way to make the changes compatible. Otherwise, we may have to deprecate the current attribute and introduce a new one.

Reply:
Done! I've updated it in the new commit. added tests in both the unit and integration tests. One test passes the value as an integer, and another passes the value as a float. All the tests are passing